### PR TITLE
Possibly fix an ICE on test

### PR DESCRIPTION
**Pull request by JohnTitor**
Fix a potential ICE on test with debug assertion, caused on rust-lang/rust#67661.
r? @oli-obk 

changelog: none

### CODE REVIEW

File: clippy_lints/src/mut_key.rs [hunk 1]
```diff
@@ -103,18 +103,23 @@ fn check_ty<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, span: Span, ty: Ty<'tcx>) {
...
         }
     }
 }
 
-fn is_concrete_type(ty: Ty<'_>) -> bool {
-    match ty.kind {
-        RawPtr(TypeAndMut { ty: inner_ty, .. }) | Ref(_, inner_ty, _) => is_concrete_type(inner_ty),
-        Dynamic(..) | Opaque(..) | Param(..) => false,
-        _ => true,
+fn concrete_type(ty: Option<Ty<'_>>) -> Option<Ty<'_>> {
```
Comment (oli-obk) [inline clippy_lints/src/mut_key.rs]: The argument does not need to be an `Option`

Comment (JohnTitor) [reply clippy_lints/src/mut_key.rs]: indeed, fixed

**Comment by JohnTitor** [comment]
Could I r=you clippy up once include this fix?

**Pull request merged**